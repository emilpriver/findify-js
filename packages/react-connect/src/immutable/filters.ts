import createRecord from './createRecord';
import { preventEvents } from '../utils/preventEvents';
import { List } from 'immutable';

const getFacetValue = (_this, type) => {
  if (type === 'range') {
    return _this.filter((_, k) => k === 'from' || k === 'to');
  }
  if (type === 'category') {
    return List([_this.get('value')]);
  }
  return _this.get('value');
};

export class Filter extends createRecord('Filter') {
  updater: any;
  queried: any;

  constructor(filter, updater) {
    super(
      filter.update('values', (values) =>
        values.map(
          (v) =>
            new FilterValue(
              v.withMutations((i) =>
                i
                  .set('type', filter.get('type'))
                  .set('name', filter.get('name'))
              ),
              updater,
              filter
            )
        )
      )
    );
    this.updater = updater;
  }

  resetValues = (e) => {
    preventEvents(e);
    const filterName = this.get('name');
    this.updater('filters', (f) => f.remove(filterName));
    return this;
  };
}

export class FilterValue extends createRecord('FilterValue') {
  updater: any;
  index: any;

  constructor(value, updater, filter) {
    super(value);
    this.updater = updater;
    this.index = filter.get('name');
  }

  toggle = (e) => {
    preventEvents(e);
    const filterName = this.index;
    const value = getFacetValue(this, this.get('type'));
    this.updater('filters', (f) =>
      f.withMutations((filters) => {
        const filter = filters.get(filterName);
        const index = filter.indexOf(value);
        return !~index
          ? filters.remove(filterName)
          : filters.removeIn([filterName, index]);
      })
    );
    return this;
  };
}
