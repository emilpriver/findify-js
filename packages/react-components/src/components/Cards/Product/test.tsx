import { fromJS } from 'immutable';
import ProductCardView from './view';
import theme from './styles.css';
import { shallow } from 'enzyme';
import { get as _get } from 'lodash';

const itemMock = fromJS({
  product_url: 'https://foo.bar',
  image_url: 'https://lorempicsum.net/100/100',
  description: 'A description for item',
  thumbnail_url: 'https://lorempicsum.net/50/50',
  title: 'Product Title',
  price: '9.99',
  reviews: { average_rating: 0 },
});

const configMock = fromJS({
  productcard: {
    title: {
      display: true,
      lines: 1,
    },
    description: {
      display: true,
      lines: 2,
    },
    price: {
      display: true,
    },
  },
  currency_config: {
    code: 'USD',
    symbol: '$',
    thousandsSeparator: ',',
    decimalSeparator: '.',
    symbolOnLeft: true,
    spaceBetweenAmountAndSymbol: false,
    decimalDigits: 2,
  },
});

describe('ProductCard view', () => {
  it('renders correctly', () => {
    expect(
      shallow(
        <ProductCardView
          theme={{}}
          item={itemMock}
          config={configMock}
          theme={theme}
        />
      )
    ).toMatchSnapshot();
  });
});
