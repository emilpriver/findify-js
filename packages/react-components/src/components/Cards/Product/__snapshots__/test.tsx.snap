// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard view renders correctly 1`] = `
<a
  className="root productCard"
  href="https://foo.bar"
>
  <div
    className="imageWrap"
  >
    <BundleAction
      item={
        Immutable.Map {
          "product_url": "https://foo.bar",
          "image_url": "https://lorempicsum.net/100/100",
          "description": "A description for item",
          "thumbnail_url": "https://lorempicsum.net/50/50",
          "title": "Product Title",
          "price": "9.99",
          "reviews": Immutable.Map {
            "average_rating": 0,
          },
        }
      }
    />
    <Image
      alt="Product Title"
      aspectRatio={1}
      className="image"
      src="https://lorempicsum.net/100/100"
      thumbnail="https://lorempicsum.net/50/50"
    />
    <div>
      <Component
        className="discountSticker"
        config={
          Immutable.Map {
            "productcard": Immutable.Map {
              "title": Immutable.Map {
                "display": true,
                "lines": 1,
              },
              "description": Immutable.Map {
                "display": true,
                "lines": 2,
              },
              "price": Immutable.Map {
                "display": true,
              },
            },
            "currency_config": Immutable.Map {
              "code": "USD",
              "symbol": "$",
              "thousand": ",",
              "decimalSeparator": ".",
              "symbolOnLeft": true,
              "spaceBetweenAmountAndSymbol": false,
              "decimalDigits": 2,
            },
          }
        }
      />
    </div>
  </div>
  <div
    className="rating"
  >
    <Rating />
  </div>
  <div
    className="variants"
  >
    Available in  variants
  </div>
  <div
    className="content"
  >
    <Title
      text="Product Title"
      theme={null}
    />
    <Description
      text="A description for item"
      theme={null}
    />
    <Price
      className="priceWrapper"
      currency={
        Object {
          "code": "USD",
          "decimalDigits": 2,
          "decimalSeparator": ".",
          "spaceBetweenAmountAndSymbol": false,
          "symbol": "$",
          "symbolOnLeft": true,
          "thousand": ",",
        }
      }
      price="9.99"
    />
    <Component
      config={
        Immutable.Map {
          "productcard": Immutable.Map {
            "title": Immutable.Map {
              "display": true,
              "lines": 1,
            },
            "description": Immutable.Map {
              "display": true,
              "lines": 2,
            },
            "price": Immutable.Map {
              "display": true,
            },
          },
          "currency_config": Immutable.Map {
            "code": "USD",
            "symbol": "$",
            "thousand": ",",
            "decimalSeparator": ".",
            "symbolOnLeft": true,
            "spaceBetweenAmountAndSymbol": false,
            "decimalDigits": 2,
          },
        }
      }
    />
  </div>
</a>
`;
